fulp_modules/features/antagonists/infiltrators/infil_objectives.dm

```diff
@@ -0,0 +1,181 @@
+	if(!employer)
+		return
+	switch(employer)
+		if(INFILTRATOR_FACTION_CORPORATE_CLIMBER)
+			var/datum/objective/assassinate/killtraitor = new
+			killtraitor.owner = owner
+			killtraitor.find_traitor_target()
+			objectives += killtraitor
+
+			var/datum/objective/escape/escape_with_identity/infiltrator/escape = new
+			escape.owner = owner
+			escape.find_sec_target()
+			objectives += escape
+
+			var/datum/objective/assassinate/killsec = new
+			killsec.owner = owner
+			killsec.target = escape.target   //assassinate the officer you're supposed to impersonate
+			killsec.update_explanation_text()
+			objectives += killsec
+
+			for(var/i = 0, i < 2, i++)
+				var/datum/objective/steal/steal_objective = new
+				steal_objective.owner = owner
+				steal_objective.find_target()
+				objectives += steal_objective
+
+		if(INFILTRATOR_FACTION_ANIMAL_RIGHTS_CONSORTIUM)
+			for(var/i = 0, i < 2, i++)
+				var/datum/objective/kill_pet/pet = new
+				pet.owner = owner
+				pet.find_pet_target()
+				objectives += pet
+
+			for(var/i = 0, i < 2, i++)
+				var/datum/objective/assassinate/kill = new
+				kill.owner = owner
+				kill.find_sci_target()
+				objectives += kill
+
+			var/datum/objective/assassinate/kill_head = new
+			kill_head.owner = owner
+			kill_head.find_head_target()
+			objectives += kill_head
+
+		if(INFILTRATOR_FACTION_GORLEX_MARAUDERS)
+			for(var/i = 0, i < rand(4,6) , i++)
+				var/datum/objective/assassinate/assassinate = new
+				assassinate.owner = owner
+				assassinate.find_target()
+				objectives += assassinate
+
+			var/datum/objective/emag_console/emag = new
+			emag.owner = owner
+			emag.update_explanation_text()
+			objectives += emag
+
+
+//Corporate Climber objectives
+
+//Find Traitor target
+/datum/objective/assassinate/proc/find_traitor_target()
+	var/list/possible_targets = list()
+	for(var/mob/living/carbon/human/player in GLOB.player_list)
+		if(player.stat == DEAD || player.mind == owner)
+			continue
+		if(player.mind?.has_antag_datum(/datum/antagonist/traitor))
+			possible_targets += player.mind
+
+	if(!possible_targets.len)
+		find_target() //if no traitors on station, this becomes a normal assassination obj
+		return
+	else
+		target = pick(possible_targets)
+
+	if(target?.current)
+		explanation_text = "Special intel has identified [target.name] the [!target_role_type ? target.assigned_role.title : target.special_role]. as a threat to Nanotrasen. Eliminate them at all costs."
+
+
+//advanced mulligan objective
+
+/datum/objective/escape/escape_with_identity/infiltrator
+	name = "escape with identity (as infiltrator)"
+
+/datum/objective/escape/escape_with_identity/infiltrator/proc/find_sec_target()
+	var/list/sec = SSjob.get_all_sec()
+	if(!sec.len)
+		find_target()
+	else
+		target = pick(sec)
+	update_explanation_text()
+
+//Animal Rights Consortium Objectives
+
+//pet killing
+
+/datum/objective/kill_pet
+	name = "Kill a command pet"
+	var/mob/living/target_pet
```

Comment:
autodoc this and put it under martyr_compatible, unique vars is supposed to be last